src/gen_smtp_client.erl

```diff
@@ -137,26 +137,24 @@ send(Email, Options, Callback) ->
 	NewOptions = lists:ukeymerge(1, lists:sort(Options),
 		lists:sort(?DEFAULT_OPTIONS)),
 	case check_options(NewOptions) of
-		ok when is_function(Callback) ->
-			spawn(fun() ->
-						process_flag(trap_exit, true),
-						Pid = spawn_link(fun() ->
-									send_it_nonblock(Email, NewOptions, Callback)
-							end
-						),
-						receive
-							{'EXIT', Pid, Reason} ->
-								case Reason of
-									X when X == normal; X == shutdown ->
-										ok;
-									Error ->
-										Callback({exit, Error})
-								end
-						end
-				end);
 		ok ->
-			Pid = spawn_link(fun () ->
-						send_it_nonblock(Email, NewOptions, Callback)
+			Pid = spawn_link(
```

Comment:
since we are already here, maybe use `proc_lib:spawn_link` for it?

Reply:
I could, but why? =^^=
I mean, `proc_lib:spawn_link` is just `erlang:spawn_link` with a cherry on top, to make it fit and play nicely in supervision trees or as a special process, neither of which this is?

Reply:
I don't have a strong opinion on this. Just have my rule of thumb to use `proc_lib` instead of raw spawns without much thinking :) But let's see what @mworrell or @arjan would say